server/controllers/chartsController/index.js

```diff
@@ -1,80 +1,156 @@
-import { ObjectID } from 'mongodb'
-
+import { ObjectId } from 'mongodb'
+import deepEqual from 'deep-equal'
 import { collections } from '../../utils/mongoCollections'
 import { SITE_NAMES } from '../../utils/siteNames'
 import {
   generateStudyTargetTotals,
   isAnyTargetIncluded,
-  N_A,
   postProcessData,
   processData,
   processTotals,
-  TOTALS_STUDY,
+  mongoQueryFromFilters,
+  sanitizeQueryParams,
 } from './helpers'
+import {
+  N_A,
+  TOTALS_STUDY,
+  STUDIES_TO_OMIT,
+  DEFAULT_CHART_FILTERS,
+  EMPTY_VALUE,
+  INCLUSION_EXCLUSION_CRITERIA_FORM,
+} from '../../constants'
 
-const STUDIES_TO_OMIT = ['files', 'combined']
-
-export const graphDataController = async (dataDb, userAccess, chart_id) => {
+export const graphDataController = async (
+  dataDb,
+  userAccess,
+  chart_id,
+  parsedQueryParams
+) => {
   const labelMap = new Map()
   const dataMap = new Map()
+  const areFiltersRequested =
+    !!Object.keys(parsedQueryParams).length &&
+    !deepEqual(DEFAULT_CHART_FILTERS, sanitizeQueryParams(parsedQueryParams))
+  const filters = areFiltersRequested
+    ? sanitizeQueryParams(parsedQueryParams)
+    : DEFAULT_CHART_FILTERS
   const chart = await dataDb
     .collection(collections.charts)
-    .findOne({ _id: ObjectID(chart_id) })
-  const allSubjects = await dataDb
-    .collection(collections.toc)
-    .find(
-      {
-        assessment: chart.assessment,
-        study: { $in: userAccess, $nin: STUDIES_TO_OMIT },
-      },
-      { projection: { collection: 1, study: 1, _id: 0 } }
-    )
-    .toArray()
-
+    .findOne({ _id: ObjectId(chart_id) })
+  const allSubjects = await getAllSubjects({
+    dataDb,
+    chart,
+    userAccess,
+    areFiltersRequested,
+    filters,
+  })
   const allowedStudies = userAccess.filter(
     (study) => !STUDIES_TO_OMIT.includes(study)
   )
   const studyTotals = generateStudyTargetTotals(chart, allowedStudies)
 
-  for await (const subject of allSubjects) {
-    const { study } = subject
-    const subjectDayData = await dataDb
-      .collection(subject.collection)
-      .find({})
-      .toArray()
+  if (allSubjects.length) {
```

Comment:
do we need this check? if `allSubjects` is an empty array that seems like it would be fine to use as we had this before.

Reply:
For some reason an empty array yields a blank page and I thought that would confuse users, by adding this it gives a message that there is no data available for the chart and it could prompt the users to select different filters, etc..

Reply:
can we look into why we get a blank page when allSubjects is empty? That is unexpected to me.

Reply:
https://github.com/AMP-SCZ/dpdash/pull/248/commits/147bf8a9965b532f34bb176e1e651f339c9bf387